tests/main/snap-mgmt/task.yaml

```diff
@@ -0,0 +1,27 @@
+summary: Check that snap-mgmt.sh works
+
+prepare: |
+    . $TESTSLIB/dirs.sh
+
+    snap install test-snapd-tools
+    snap list | MATCH test-snapd-tools
+    before=$(find ${SNAP_MOUNT_DIR} -type d | wc -l)
+    if [ "$before" -lt 2 ]; then
+        echo "${SNAP_MOUNT_DIR} empty - test setup broken"
+        exit 1
+    fi
+
+execute: |
+    . $TESTSLIB/dirs.sh
+
+    echo "A purge will really purge things"
+    $LIBEXECDIR/snapd/snap-mgmt --purge
+
+    for d in ${SNAP_MOUNT_DIR} /var/snap; do
```

Comment:
I think we also need to verify that:
* snap images are not mounted
* directories that may break snapd if that is reinstalled are properly removed (I know of /var/lib/snapd/assertions/* /var/lib/snapd/cookie/*) (this was the primary motivation for #4363)
* probably it's best if all directories under /var/lib/snapd/ are empty